src/vmm/src/vstate/vcpu/mod.rs

```diff
@@ -639,29 +639,152 @@ mod tests {
 
     lazy_static! {
         static ref RESULT: Mutex<Option<std::result::Result<VcpuExit<'static>, errno::Error>>> =
-            Mutex::new(Some(Ok(VcpuExit::InternalError)));
+            Mutex::new(None);
     }
 
     impl Vcpu {
         pub fn emulate(&self) -> std::result::Result<VcpuExit, errno::Error> {
-            RESULT.lock().unwrap().take().unwrap()
+            let mut g = RESULT.lock().unwrap();
+            if (&*g).is_some() {
+                return g.take().unwrap();
+            }
+            Err(errno::Error::new(libc::SIGILL))
         }
     }
 
+    struct DummyDevice;
+    impl devices::BusDevice for DummyDevice {}
+
     #[test]
     fn test_run_emulation() {
-        let (_vm, vcpu, _vm_mem) = setup_vcpu(0x1000);
+        let (_vm, mut vcpu, _vm_mem) = setup_vcpu(0x1000);
+
+        {
+            let mut g = RESULT.lock().unwrap();
+            *g = Some(Ok(VcpuExit::Hlt));
+        }
```

Comment:
This block is repeated multiple times, it can move to a func or macro

```rust
fn set_emu_ok_result(ret: VcpuExit) {
    let mut g = RESULT.lock().unwrap();
    *g = Some(Ok(ret));
}

fn set_emu_err_result(ret: VcpuExit) {
    let mut g = RESULT.lock().unwrap();
    *g = Some(Err(ret));
}
```